paymentsheet/src/main/java/com/stripe/android/paymentelement/confirmation/DefaultConfirmationHandler.kt

```diff
@@ -44,9 +44,12 @@ internal class DefaultConfirmationHandler(
             coroutineScope.launch {
                 delay(1.seconds)
 
+                val currentValue = _state.value
+
                 if (
-                    _state.value is ConfirmationHandler.State.Confirming &&
-                    isAwaitingForResultData?.receivesResultInProcess != true
+                    currentValue is ConfirmationHandler.State.Confirming &&
+                    currentValue.option == isAwaitingForResultData?.confirmationOption &&
```

Comment:
What case is this? Do we have tests?

Reply:
Realized that when coming back, we can have a `NextStep` result that occurs from a launched confirmation activity that isn't handled in the process which can keep the state confirming so this would cancel the flow early when returning from process death. This isn't happening currently in production because all the actions that currently produce `NextStep` all receive their result in process but `Link` does not (full screen activity).

I've now realized this also doesn't work though because if we other `NextStep` producing definitions that don't change the confirmation option, this can also fail. This can be solved by removing the data when the result is received.

Reply:
I'm going to fix & merge this in a separate PR first actually then rebase this PR on the change.